### PR TITLE
Adding snippet and a GIF demo about dash-embedded in /integrating-dash

### PR DESCRIPTION
**Pull request by xhlulu**
gif: 
embd_state_share

It seems that there's some some issues with JS highlighting; does anyone have an idea?

@harryturr thoughts on the framing and adding snippets?

---

Post-merge checklist:

The master branch is auto-deployed to `dash.plotly.com`.
Once you have merged your PR, wait 5-10 minutes and check dash.plotly.com
to verify that your changes have been made.

- [ ] I understand

If this PR documents a new feature of Dash:

- [ ] Comment on the original Dash issue with a link to the new docs.
- [ ] Reply to any community thread(s) asking for this feature.

If this PR includes a new dataset available at a remote URL:
- [ ] I have added this dataset to the `datasets/` folder
- [ ] I have added a mapping between the remote URL and the filename in the
`datasets/` folder into the `find_and_replace` dict in `dash_docs/tools.py`

If this PR adds an image or animated GIF:
- [ ] This image was saved and referenced locally rather than via an external link

If I introduced a new relative link inside `dcc.Markdown`:
- [ ] I considered whether I could replace the `dcc.Markdown` call with `rc.Markdown`, which will replace relative links with `tools.relpath` internally. Otherwise, I used e.g. `<dccLink href=tools.relpath('/layout') children="the first chapter"/>` instead of `[the first chapter](/layout)` (importing `tools` from `dash_docs`), and set `dangerously_allow_html=true` in the `dcc.Markdown` call.

If I changed the `chapter_index` by removing or relocating a page:
- [ ] I added a redirect in `dash_docs/server.py` from the old URL to the new URL

### CODE REVIEW

File: dash_docs/chapters/integrating_dash/index.py [hunk 1]
```diff
@@ -48,6 +48,83 @@
...
+
+    function App() {
+        return (
+            <div style={...}>
+                ...
+                <h1>Embedded Dash Application</h1>
+                <DashApp
+                    config={{
+                        url_base_pathname: "http://dash.tests:8050",
+                    }}
```
Comment (harryturr) [inline dash_docs/chapters/integrating_dash/index.py:121]:
```suggestion
                    }}
                    value={this.state.sharedData}
```

File: dash_docs/chapters/integrating_dash/index.py [hunk 2]
```diff
@@ -48,6 +48,83 @@
     `https://<your-dash-enterprise-hostname>/Docs/embedded-middleware`**,
     replacing `<your-dash-enterprise-hostname>` with the hostname of your
     licensed Dash Enterprise in your VPC. [Look up the hostname for your company's license](https://go.plotly.com/company-lookup)
+    
+    #### Sharing State between a Javascript Host app and an embedded Dash App
+
+    You can find here a simple example illustrating multidirectional shared state, enabling the communication between your JavaScript host app and your Dash app.
+
+    ![GIF showing how to use dash embedded](https://raw.githubusercontent.com/plotly/dash-docs/master/images/dash-embedded-js-host.gif)
```
Comment (harryturr) [inline dash_docs/chapters/integrating_dash/index.py:56]: image
I don't see the gif in the review app

File: dash_docs/chapters/integrating_dash/index.py (hunk 1 above)
Comment (xhlulu) [reply dash_docs/chapters/integrating_dash/index.py]: Ok actually it seems it's using the class API, so I'll just change function to class and move it to a `render` function

**Comment by xhlulu** [comment]
I just changed the function into the correct class; let me know if the new version is OK

File: dash_docs/chapters/integrating_dash/index.py [hunk 3]
```diff
@@ -48,6 +48,81 @@
...
+    If you are using a React app, you can import the component and use it inside JSX:
+    ```js
+    import { DashApp } from "dash-embedded-component";
+
+    window.React = React;
+    window.ReactDOM = ReactDOM;
+    window.PropTypes = PropTypes;
+
+    function App() {
+        return (
```
Comment (harryturr) [inline dash_docs/chapters/integrating_dash/index.py]:
```suggestion
    function App() {
      const [exampleState, setExampleState] = useState(
          {sharedData: {
              ....
          }
      )
        return (
```

File: dash_docs/chapters/integrating_dash/index.py (hunk 1 above)
Comment (harryturr) [reply dash_docs/chapters/integrating_dash/index.py]: Sorry, the `this.state.shareData` would be a local state defined in a class object. 

So here maybe you can just add a dummy state variable at the top of our function and pass that as the `value` prop.

**Review by harryturr — APPROVED** [review]
@xhlulu that looks good to me, that way we can see the relationship between Consumer components and host app. But if you prefer functional components feel free

**Comment by xhlulu** [comment]
@harryturr Thanks! I think `class` is fine.

@chriddyp anything missing?

File: dash_docs/chapters/integrating_dash/index.py [hunk 4]
```diff
@@ -48,6 +48,100 @@
...
+                        id="host-app-multiply", path=["myObject", "multiplyFunc"]
+                    ),
+                    ConsumerFunction(id="host-app-sum", path=["myObject", "sumFunc"]),
+                    ddk.ControlItem(...),
+                    ...
+                ],
+                width=50,
+            ),
+            ...
+        ]
```
Comment (chriddyp) [inline dash_docs/chapters/integrating_dash/index.py:140]: I'd recommend putting in the callback examples here too

Comment (xhlulu) [reply dash_docs/chapters/integrating_dash/index.py:140]: Awesome! I just pushed some truncated callbacks. Let me know if that's what you had in mind

**Comment by xhlulu** [comment]
@chriddyp is this ready to be merged?

**Review by chriddyp — APPROVED** [review]
💃🏼

**Pull request merged**